src/symmetric_crypto/key_wrapping_rfc_5649.rs

```diff
@@ -0,0 +1,498 @@
+use aead::{generic_array::GenericArray, KeyInit};
+use aes_gcm::aes::{Aes128, Aes192, Aes256};
+use chacha20::cipher::{BlockDecrypt, BlockEncrypt};
+
+use crate::CryptoCoreError;
+
+#[allow(dead_code)]
+const DEFAULT_IV: u64 = 0xA6A6_A6A6_A6A6_A6A6;
+#[allow(dead_code)]
+const DEFAULT_RFC5649_CONST: u32 = 0xA659_59A6_u32;
+
+/// Build the iv according to the RFC 5649
+///
+/// `data_size` is the size of the key to wrap
+#[allow(dead_code)]
+fn build_iv(data_size: usize) -> u64 {
+    let l = u64::from(DEFAULT_RFC5649_CONST);
+    let r = u32::to_le(data_size as u32);
+
+    l << 32 | u64::from(r)
+}
+
+/// Check if `iv` value is appropriate according to the RFC 5649
+#[allow(dead_code)]
+fn check_iv(iv: u64, data: &[u8]) -> bool {
+    let data_size: usize = data.len();
+    if (iv >> 32) as u32 != DEFAULT_RFC5649_CONST {
+        return false;
+    }
+
+    let real_data_size = u32::to_le(iv as u32) as usize;
+    if real_data_size > data_size || real_data_size <= (data_size - 8) {
+        return false;
+    }
+
+    data[real_data_size..].iter().all(|&x| x == 0)
+}
+
+/// Wrap a plain text of variable length
+///
+/// Follows RFC 5649
+/// KEK stands for `Key-Encryption Key`
+/// The function name matches the one used in the RFC and has no link to the
+/// unwrap function in Rust
+#[allow(dead_code)]
+pub fn wrap(plain: &[u8], kek: &[u8]) -> Result<Vec<u8>, CryptoCoreError> {
+    let n = plain.len();
+    let m = n % 8;
+
+    let padded_plain = if m != 0 {
+        // Pad the plaintext octet string on the right with zeros
+        let missing_bytes = 8 - m;
+        [plain.to_vec(), vec![0u8; missing_bytes]].concat()
+    } else {
+        plain.to_vec()
+    };
```

Comment:
You can avoid a clone when padding and avoid branching:
```suggestion
    let padded_plain = [plain, &vec![0u8; 8 - m]].concat();
```